src/main/java/org/myrobotlab/service/config/RuntimeConfig.java

```diff
@@ -32,8 +38,20 @@ synchronized public void add(String name) {
     registry.add(name);
   }
 
-  public void remove(String name) {
-    registry.remove(name);
+  public boolean remove(String name) {
+    return registry.remove(name);
+  }
+  
+  public void removeStartsWith(String startsWith) {
+    ArrayList<String> remove = new ArrayList<>();
+    for (String n: registry) {
```

Comment:
Can be collapsed down to one while-loop with an iterator, that way removing an item doesn't end up skipping elements or anything. Would need to synchronize on it to prevent `ConcurrentModificationException`s though, which is probably what we want anyway, unless we can guarantee single-threaded access

Reply:
added synchronized, I don't want to change the datatype as I've had issues in the past serializing thread safe versions ...
Don't want to reduce it down to 1 loop, as this is a simple way not to trip on removed items..
Heh, for the record ChatGPT refactored it this way
```
public void removeStartsWith(String startsWith) {
    List<String> remove = registry.stream()
                                   .filter(n -> n.startsWith(startsWith))
                                   .collect(Collectors.toList());
    registry.removeAll(remove);
}

```
Which is more dense, maybe a minuscule more efficient, but I find it difficult to read ... me being a dinosaur

Reply:
Going off of ChatGPT's suggestion, this could actually be reduced down to a single line:
`registry.removeIf(n -> n.startsWith(startsWith));`

In my opinion, this is the easiest way to read it. I actually misread the original version, I thought it was doing the removal within the first for loop which could result in skipping elements, but aggregating them all in one list and then removing all of the elements from that list should work if you wanted to keep it.

Reply:
updated :laughing: